rapid7_insightidr/komand_rapid7_insightidr/actions/query/action.py

```diff
@@ -0,0 +1,57 @@
+import komand
+from .schema import QueryInput, QueryOutput, Input, Output, Component
+# Custom imports below
+from komand.exceptions import PluginException
+from komand_rapid7_insightidr.util.endpoints import QueryLogs
+from komand_rapid7_insightidr.util.resource_helper import ResourceHelper
+import json
+import time
+
+
+class Query(komand.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name='query',
+            description=Component.DESCRIPTION,
+            input=QueryInput(),
+            output=QueryOutput())
+
+    def run(self, params={}):
+        time_now = int(time.time())
+        request = ResourceHelper(self.connection.session, self.logger)
+        request_params = {
+            "from": (time_now - 7776000) * 1000,
+            "to": time_now * 1000
+        }
+        response = request.resource_request(
+            QueryLogs.get_query_logs(self.connection.url, params.get(Input.ID)),
+            'get',
+            params=request_params
+        )
+
+        try:
+            result = json.loads(response['resource'])
+            if response["status"] == 202:
```

Comment:
Need to catch other status codes here

Reply:
Why? In documentation only 200 and 202 are used. Other codes (200-299) are treted as 200. Other statuses (<200 and >=300) are catched in `ResourceHelper` class.